Mage/src/main/java/mage/abilities/AbilityImpl.java

```diff
@@ -869,19 +865,46 @@ public void addCost(Cost cost) {
         } else {
             // as single cost
             if (cost instanceof ManaCost) {
-                this.addManaCost((ManaCost) cost);
+                addManaCost((ManaCost) cost);
             } else {
-                this.costs.add(cost);
+                if (costs == null) {
+                    costs = new CostsImpl<>();
+                }
+                costs.add(cost);
             }
         }
     }
 
     @Override
-    public void addManaCost(ManaCost cost) {
-        if (cost != null) {
-            this.manaCosts.add(cost);
-            this.manaCostsToPay.add(cost);
+    public void addManaCostsToPay(ManaCost manaCost) {
+        if (manaCost == null) {
+            return;
+        }
+        if (manaCostsToPay == null) {
+            manaCostsToPay = new ManaCostsImpl<>();
         }
+        if (manaCost instanceof ManaCosts) {
+            manaCostsToPay.addAll((ManaCosts) manaCost);
+        } else {
+            manaCostsToPay.add(manaCost);
+        }
+    }
+
+    @Override
+    public void addManaCost(ManaCost manaCost) {
```

Comment:
TODO after this PR merged: switch a few individual card usages over to addCost like everything else